src/core/Modal/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { ReactNode, useRef, useState } from 'react';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { axeTest } from '../../../utils/test/axe';
+
+import { Modal, ModalProps } from './Modal';
+import { ModalContent, ModalFooter, ModalTitle } from '../';
+import { Button } from '../../../';
+
+describe('Basic modal', () => {
+  const text = 'Modal Content';
+
+  const BasicModal = (props?: Partial<ModalProps>) => (
+    <Modal visible={true} usePortal={false} {...props}>
+      <ModalContent>
+        <ModalTitle>Test modal</ModalTitle>
+        <p>{text}</p>
+      </ModalContent>
+      <ModalFooter>
+        <Button>OK</Button>
+      </ModalFooter>
+    </Modal>
+  );
+
+  it('should be hidden when visible is set to false', () => {
+    const { container } = render(BasicModal({ visible: false }));
+    expect(container.firstChild).toBe(null);
+  });
+
+  it('should have given className', () => {
+    const { container } = render(BasicModal({ className: 'test-class-name' }));
+    expect(container.firstChild).toHaveClass('test-class-name');
+  });
+
+  it('should match snapshot', () => {
+    const { container } = render(BasicModal());
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('should not have basic accessibility issues', axeTest(BasicModal()));
+});
+
+describe('Modal variant', () => {
+  const ModalWithProps = (props?: Partial<ModalProps>) => (
+    <Modal visible={true} usePortal={false} {...props}>
+      <ModalContent>
+        <ModalTitle>Test modal</ModalTitle>
+        <p>Modal Content</p>
+      </ModalContent>
+      <ModalFooter>
+        <Button>OK</Button>
+        <Button variant="secondary">Cancel</Button>
+      </ModalFooter>
+    </Modal>
+  );
+
+  it('smallScreen should have correct classname', () => {
+    const { container } = render(ModalWithProps({ variant: 'smallScreen' }));
+    expect(container.firstChild).toHaveClass('fi-modal--small-screen');
+  });
+});
+
+describe('Modal focus', () => {
+  const ModalWithProps = (children: ReactNode, props?: Partial<ModalProps>) => (
+    <Modal visible={true} usePortal={false} {...props}>
+      <ModalContent>
+        <ModalTitle>Test modal</ModalTitle>
+        {children}
+      </ModalContent>
+      <ModalFooter>
+        <Button>OK</Button>
+        <Button variant="secondary">Cancel</Button>
+      </ModalFooter>
+    </Modal>
+  );
+
+  it('should be on primary button if there is no other focusable content', async () => {
+    const { getAllByRole } = render(ModalWithProps(<p>Modal Content</p>));
+    await waitFor(() => expect(getAllByRole('button')[0]).toHaveFocus());
```

Comment:
This is a neat selector :) I learned something.

Reply:
New for me as well, but waitFor is neat util since it allows testing of asynchronous event pretty easily.